source/module_base/test/timer_test.cpp

```diff
@@ -0,0 +1,140 @@
+#include "../timer.h"
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include <fstream>
+#include <cstdio>
+#ifdef __MPI
+#include "mpi.h"
+#endif
+
+/************************************************
+ *  unit test of class timer
+ ***********************************************/
+
+/**
+ * - Tested Functions:
+ *   - Tick
+ *     - tick 1st time, set start_flag to false
+ *     - tick 2nd time, calculate time duration
+ *   - Start
+ *     - start total time calculation
+ *   - PrintAll
+ *     - print computational processes with time > 0.1 s 
+ *   - Finish
+ *     - finish total time calculation 
+ *     - print computational processes with time > 0.1 s 
+ *   - PrintUntilNow
+ *     - stop total time calculation
+ *     - print total time until now
+ *     - then start total time calculation again
+ */
+
+class TimerTest : public testing::Test
+{
+protected:
+	//
+	// for capturing stdout
+	std::string output;
+	// for output in file
+	std::ofstream ofs;
+	std::ifstream ifs;
+	void TearDown()
+	{
+		remove("tmp");
+	}
+};
+
+
+TEST_F(TimerTest, Tick)
+{
+	ModuleBase::timer::tick("wavefunc","evc");
+	// after 1st call of tick, start_flag becomes false
+	EXPECT_FALSE(ModuleBase::timer::timer_pool["wavefunc"]["evc"].start_flag);
+	sleep(0.1); // ms
```

Comment:
Where is the definition of this function? I suggest using [`sleep_for`](https://www.cplusplus.com/reference/thread/this_thread/sleep_for/).